bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ObservationWeatherHandler.java

```diff
@@ -0,0 +1,193 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.unit.MetricPrefix;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.fmiweather.internal.client.Data;
+import org.openhab.binding.fmiweather.internal.client.FMIResponse;
+import org.openhab.binding.fmiweather.internal.client.FMISID;
+import org.openhab.binding.fmiweather.internal.client.Location;
+import org.openhab.binding.fmiweather.internal.client.ObservationRequest;
+import org.openhab.binding.fmiweather.internal.client.exception.FMIResponseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ObservationWeatherHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Sami Salonen - Initial contribution
+ */
+@NonNullByDefault
+public class ObservationWeatherHandler extends AbstractWeatherHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ObservationWeatherHandler.class);
+    private static final long OBSERVATION_LOOK_BACK_SECONDS = TimeUnit.MINUTES.toSeconds(30);
+    private static final int STEP_MINUTES = 10;
+    private static final int POLL_INTERVAL_SECONDS = 600;
+
+    public static final Unit<Length> MILLIMETRE = MetricPrefix.MILLI(METRE);
+    public static final Unit<Length> CENTIMETRE = MetricPrefix.CENTI(METRE);
+
+    private static final Map<String, Map.Entry<String, @Nullable Unit<?>>> CHANNEL_TO_OBSERVATION_FIELD_NAME_AND_UNIT = new HashMap<>(
+            11);
+
+    private static void addMapping(String channelId, String requestField, @Nullable Unit<?> unit) {
+        CHANNEL_TO_OBSERVATION_FIELD_NAME_AND_UNIT.put(channelId,
+                new AbstractMap.SimpleImmutableEntry<>(requestField, unit));
+    }
+
+    static {
+        addMapping(CHANNEL_TEMPERATURE, PARAM_TEMPERATURE, CELSIUS);
+        addMapping(CHANNEL_HUMIDITY, PARAM_HUMIDITY, PERCENT);
+        addMapping(CHANNEL_WIND_DIRECTION, PARAM_WIND_DIRECTION, DEGREE_ANGLE);
+        addMapping(CHANNEL_WIND_SPEED, PARAM_WIND_SPEED, METRE_PER_SECOND);
+        addMapping(CHANNEL_GUST, PARAM_WIND_GUST, METRE_PER_SECOND);
+        addMapping(CHANNEL_PRESSURE, PARAM_PRESSURE, MILLIBAR);
+        addMapping(CHANNEL_PRECIPITATION_AMOUNT, PARAM_PRECIPITATION_AMOUNT, MILLIMETRE);
+        addMapping(CHANNEL_SNOW_DEPTH, PARAM_SNOW_DEPTH, CENTIMETRE);
+        addMapping(CHANNEL_VISIBILITY, PARAM_VISIBILITY, METRE);
+        addMapping(CHANNEL_CLOUDS, PARAM_CLOUDS, null);
+        addMapping(CHANNEL_OBSERVATION_PRESENT_WEATHER, PARAM_PRESENT_WEATHER, null);
+    }
+
+    private @NonNullByDefault({}) String fmisid;
```

Comment:
I have understood that the way it is know is recommended by the [conventions](https://www.openhab.org/v2.4/docs/developer/development/conventions.html): 

> Fields within ThingHandler classes that are initialized within the initialize() method may also be annotated like this, because the framework ensures that initialize() will be called before any other method.